libnavui-dropin/src/main/java/com/mapbox/navigation/dropin/component/recenter/RecenterButtonBehaviour.kt

```diff
@@ -0,0 +1,57 @@
+package com.mapbox.navigation.dropin.component.recenter
+
+import androidx.lifecycle.asFlow
+import com.mapbox.maps.CameraOptions
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.dropin.component.camera.DropInCameraMode
+import com.mapbox.navigation.dropin.component.camera.DropInCameraState
+import com.mapbox.navigation.dropin.component.location.DropInLocationState
+import com.mapbox.navigation.dropin.lifecycle.UIComponent
+import com.mapbox.navigation.utils.internal.toPoint
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.asSharedFlow
+import kotlinx.coroutines.flow.asStateFlow
+
+internal class RecenterButtonBehaviour(
+    private val cameraState: DropInCameraState,
+    private val locationState: DropInLocationState,
+    private val shouldShowButton: VisibilityPolicy = VisibilityPolicy.WHEN_NOT_FOLLOWING
+) : UIComponent() {
+
+    private val _isButtonVisible = MutableStateFlow(false)
+    val isButtonVisible = _isButtonVisible.asStateFlow()
+
+    private val _moveCameraEvent = MutableSharedFlow<CameraOptions>(
```

Comment:
Are you planning to change the behavior as we discussed today morning?

Reply:
Not in this PR. I was planning on doing it in PR for this https://github.com/mapbox/navigation-sdks/issues/1445

Reply:
@abhishek1508 Looks like after rebasing on top of main I had to refactor this anyway.
I removed `_moveCameraEvent` from `RecenteButtonBehaviour`  and added temporary method `requestCameraUpdate(event: CameraUpdateEvent) ` to `DropInCameraState`.
We will revisit this when working on [#1445](https://github.com/mapbox/navigation-sdks/issues/1445).